Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -102,6 +102,34 @@ ScrollTimeline::ScrollTimeline(Scroller scroller, ScrollAxis axis)
     m_scroller = scroller;
 }
 
+Element* ScrollTimeline::source() const
+{
+    switch (m_scroller) {
+    case Scroller::Nearest: {
+        if (CheckedPtr subjectRenderer = m_source->renderer()) {
+            if (RefPtr source = subjectRenderer->enclosingScrollableContainer()->element()) {
+                auto document = source->protectedDocument();
+                // RenderObject::enclosingScrollableContainer() will return the document element even in
+                // quirks mode, but the scrolling element in that case is the <body> element, so we must
+                // make sure to return Document::scrollingElement() in case the document element is
+                // returned by enclosingScrollableContainer() but it was not explicitly set as the source.
+                RefPtr documentElement = document->documentElement();
+                if (m_source.get() != documentElement && source == documentElement)
```

Comment:
Confusing to have `source` and `m_source`.

Reply:
I still find this code a bit confusing. Can you have an early return for the non-root case, and deal with the documentElement/scrollingElement case in its own code block?

Reply:
> Confusing to have `source` and `m_source`.

Indeed! I'll rename `source` to `nearestSource` since this is what we're looking for here.

Reply:
> I still find this code a bit confusing. Can you have an early return for the non-root case, and deal with the documentElement/scrollingElement case in its own code block?

Will try in an updated version of this patch, let me know if you have further comments.